File: packages/restapi/src/lib/chat/getGroup.ts
import { getAPIBaseUrls } from '../helpers';
import Constants, { ENV } from '../constants';
import { GroupDTO } from '../types';
import { axiosGet } from '../utils/axiosUtil';
import { handleError } from '../errors/ValidationError';

/**
 *  GET /v1/chat/groups/:chatId
 */

export interface GetGroupType {
  chatId: string;
  env?: ENV;
}

export const getGroup = async (options: GetGroupType): Promise<GroupDTO> => {
  const { chatId, env = Constants.ENV.PROD } = options || {};
  try {
    if (chatId == null || chatId.length == 0) {
      throw new Error(`chatId cannot be null or empty`);
    }

    const API_BASE_URL = getAPIBaseUrls(env);
    const requestUrl = `${API_BASE_URL}/v1/chat/groups/${chatId}`;
    const response = await axiosGet<GroupDTO>(requestUrl);
    return response.data;
  } catch (err) {
    throw handleError(err, getGroup.name);
  }
};

File: packages/push-api-dart/lib/src/chat/src/get_group.dart
import 'package:push_api_dart/push_api_dart.dart';

Future<GroupDTO?> getGroup({required String chatId}) async {
  if (chatId.isEmpty) {
    throw Exception('chatId cannot be null or empty');
  }

  final result = await http.get(path: '/v1/chat/groups/$chatId');

  if (result == null) {
    return null;
  }
  return GroupDTO.fromJson(result);
}
